ods_ci/utils/scripts/fetch_tests.py

```diff
@@ -20,39 +21,58 @@ def get_repository(test_repo):
     If $test_repo is a local path, the function checks the path exists.
     """
     repo_local_path = "./ods_ci/ods-ci-temp"
+    cloned = False
     if "http" in test_repo or "git@" in test_repo:
         print("Cloning repo ", test_repo)
-        ret = execute_command(f"git clone {test_repo} {repo_local_path}")
-        if "error" in ret.lower():
-            # actual error gets printed during "execute_command"
-            raise Exception("Failed to clone the given repository")
+        cloned = True
+        execute_command(f"git clone {test_repo} {repo_local_path}")
     elif not os.path.exists(test_repo):
         raise FileNotFoundError(f"local path {test_repo} was not found")
     else:
         print("Using local repo ", test_repo)
         repo_local_path = test_repo
-    return repo_local_path
+    return repo_local_path, cloned
 
 
 def checkout_repository(ref):
     """
     Checkouts the repository at current directory to the given branch/commit ($ref)
     """
-    ret = execute_command(f"git checkout {ref}")
-    if "error" in ret.lower():
-        # actual error gets printed during "execute_command"
-        raise Exception(f"Failed to checkout to the given branch/commit {ref}")
-    ret = execute_command("git checkout")
+    execute_command(f"git checkout {ref}")
+    execute_command("git checkout")
 
 
-def extract_test_cases_from_ref(repo_local_path, ref):
+def get_branch(ref_to_exclude, selector_attribute):
+    """
+    List the remote branches and sort by last commit date (ASC order), exclude $ref_to_exclude and get latest
+    """
+    ref_to_exclude_esc = ref_to_exclude.replace("/", r"\/")
+    if "master" in ref_to_exclude or "main" in ref_to_exclude:
+        cmd = f"git branch -r --sort={selector_attribute} | grep releases/"
+    else:
+        cmd = rf"git branch -r --sort={selector_attribute} | grep releases/  | sed  's/.*{ref_to_exclude_esc}$/current/g' |  grep -zPo '[\S\s]+(?=current)'"
+    ret = execute_command(cmd)
+    branches = ret.split(" ")
+    ret = branches[-1].split("\x00")[0].strip().replace("\n", "")
+    if ret != "":
+        print(f"Done. {ret} branch selected as ref_2")
+        return ret
+    else:
+        raise Exception("Failed to auto-selecting ref_2 branch.")
```

Comment:
we will get to this line only if the ret string is empty 
otherwise, the return from the last line will occur 
```suggestion
    raise Exception("Failed to auto-selecting ref_2 branch.")
```

Reply:
I changed it slightly:

```code
if not branch or "fatal:" in branch:
        raise Exception("Failed to auto-selecting ref_2 branch.")
 print(f"Done. {branch} branch selected as ref_2")
 return branch
```

It's still wrong to do a error check like this one, we should rather check the return code of the command. But the function we have "execute_command" doesn-t return it and I'd prefer not enlarging the scope of the PR so much.